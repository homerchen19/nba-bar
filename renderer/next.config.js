const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
