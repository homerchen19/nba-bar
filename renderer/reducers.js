import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
