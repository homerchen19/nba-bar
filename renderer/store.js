import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
