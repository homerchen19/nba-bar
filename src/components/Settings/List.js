import React from 'react';
import PropTypes from 'prop-types';
import { Flex, List as AntdList } from 'antd-mobile';
import styled from 'styled-components';

const Wrapper = styled(Flex)`
  width: 100vw;
  overflow-y: scroll !important;

  ::-webkit-scrollbar {
    display: none;
  }
`;

const StyledList = styled(AntdList)`
  width: 100%;
`;

const StyledItem = styled(AntdList.Item)`
  cursor: pointer;
`;

const List = ({ openExternal, quit }) => (
  <Wrapper direction="column" justify="start" align="center">
    <StyledList>
      <StyledItem
        onClick={openExternal('https://github.com/xxhomey19/nba-bar')}
      >
        About NBA Bar
      </StyledItem>
      <StyledItem
        onClick={openExternal('https://github.com/xxhomey19/nba-bar/issues')}
      >
        Report issues
      </StyledItem>
      <StyledItem extra="âŒ˜Q" onClick={quit}>
        Quit NBA Bar
      </StyledItem>
    </StyledList>
  </Wrapper>
);

List.propTypes = {
  openExternal: PropTypes.func.isRequired,
  quit: PropTypes.func.isRequired,
};

export default List;
