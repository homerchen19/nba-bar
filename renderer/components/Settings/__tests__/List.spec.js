import React from 'react';
import renderer from 'react-test-renderer';

import List from '../List';

it('be defined', () => {
  expect(List).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    quit: jest.fn(),
    openExternal: jest.fn(),
  };
  const tree = renderer.create(<List {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
