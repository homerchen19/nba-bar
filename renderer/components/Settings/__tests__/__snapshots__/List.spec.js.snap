// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox List__Wrapper-pnwga2-0 jXdhXV am-flexbox-dir-column am-flexbox-justify-start am-flexbox-align-center"
>
  <div
    className="am-list List__StyledList-pnwga2-1 jDtvWZ"
  >
    <div
      className="am-list-body"
    >
      <div
        className="am-list-item List__StyledItem-pnwga2-2 heegGh am-list-item-middle"
        onClick={[Function]}
      >
        <div
          className="am-list-line"
        >
          <div
            className="am-list-content"
          >
            About NBA Bar
          </div>
        </div>
        <div
          className="am-list-ripple"
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
      <div
        className="am-list-item List__StyledItem-pnwga2-2 heegGh am-list-item-middle"
        onClick={[Function]}
      >
        <div
          className="am-list-line"
        >
          <div
            className="am-list-content"
          >
            Report issues
          </div>
        </div>
        <div
          className="am-list-ripple"
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
      <div
        className="am-list-item List__StyledItem-pnwga2-2 heegGh am-list-item-middle"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="am-list-line"
        >
          <div
            className="am-list-content"
          >
            Quit NBA Bar
          </div>
          <div
            className="am-list-extra"
          >
            âŒ˜Q
          </div>
        </div>
        <div
          className="am-list-ripple"
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
